docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,225 @@
+kind: AdoptedResource
+metadata:
+  name: adopted-data-bucket
+spec:
+   apiVersion: s3.services.k8s.aws/v1alpha1
+   kind: Bucket
+   metadata:
+        name: data-bucket
+        namespace: team-a
+        annotations:
+            ...
+   identifier:
+        arn: arn:aws:s3:::my-data-bucket
+```
+
+**3. An adopted API Gateway v2 API (using an ID identifier)**
+
+```
+apiVersion: apigateway.services.k8s.aws/v1alpha1
+kind: AdoptedResource
+metadata:
+  name: adopted-api-gateway
+spec:
+   apiVersion: apigateway.services.k8s.aws/v1alpha1
+   kind: Api
+   identifier:
+        id: 123456789012
+```
+
+### Custom Resource Validation Webhook
+
+In order to minimise the chance of user’s facing unexpected behaviours, a validating webhook will validate any adopted resource changes prior to submission to the ACK managers. The webhook will primarily reject the following error cases:
+
+1. The provided target ACK type does not exist within the cluster
+2. The target resource would override an existing resource
+3. The target resource type cannot be adopted
+
+## Adopted K8s Resource Tagging
+
+All adopted resources will be annotated with an ACK-specific annotation that differentiates it from a natively created resource. This annotation serves to ensure clarity for users about possible specification differences introduced during the adoption process, or for filtering by adopted status. However, once adopted into a cluster, a resource will be reconciled with the same logic as any natively created resource. 
+
+The proposed k8s adopted tag is `services.k8s.aws/adopted: true`
+
+The adopted annotation would remain on the custom resource for the remainder of its lifecycle, and should not be modified by any other manager operation. 
+
+## AWS Resource Tagging
+
+AWS resources adopted by an ACK controller will have tags retroactively applied (whenever possible) to indicate its association with an ACK deployment. The details of these tags are discussed in another issue ([#148](https://github.com/aws/aws-controllers-k8s/issues/148)). ACK managers should not place any conditions on the existence of ACK tags on the AWS resources to gate the adoption process. The manager should should override any existing ACK tags.
+
+## Resource Type Blocklisting
+
+Adoption, for some resources, is not possible due to any number of the following reasons:
+
+* They are ephemeral resources which don’t support additional lifecycle operations (one-time jobs)
+* The service cannot provide adequate information to produce a full specification
+* Unforeseen limitations within ACK or the service directly
+
+For any of these cases, the resource can be marked as “not adoptable” and the validation webhook will subsequently reject any adopted resources of this type. Therefore, the manager will never attempt to run the adoption process. This flag can be set in the generator configuration file with the `cannot_adopt` boolean option supplied for any resource.
```

Comment:
++ nice.

Reply:
I still don't know if I like this boolean name. Some other ideas:
- `is_adoptable`
- `block_adoption`

Reply:
`is_adoptable` would be fine. I prefer it, actually, versus `cannot_adopt` because I prefer boolean attributes to be positive versus negative default. `can_adopt` would also work.